File: PythonActivities/Project_Activity6.py
import time

from selenium import webdriver
from selenium.webdriver.common.by import By

def Project_Activity6():
    driver = webdriver.Chrome()
    driver.get("https://alchemy.hguy.co/lms/")
    myAccountLink = driver.find_element(By.XPATH,"//a[@href='https://alchemy.hguy.co/lms/my-account/']").click()
    time.sleep(10)
    myAccountTitle = driver.title
    assert myAccountTitle =="My Account – Alchemy LMS"
    print(myAccountTitle)

    firstloginButton = driver.find_element(By.XPATH,"//div/a[@class='ld-login ld-login ld-login-text ld-login-button ld-button']").click()
    time.sleep(10)
    userNameTextBox = driver.find_element(By.ID,"user_login").send_keys("root")
    passwordTextBox = driver.find_element(By.ID,"user_pass").send_keys('pa$$w0rd')
    loginButton = driver.find_element(By.XPATH,"//p/input[@class='button button-primary']").click()
    time.sleep(10)
    userDisplayName = driver.find_element(By.XPATH,"(//span[@class='display-name'])[1]").text
    assert userDisplayName == "root"

    print(userDisplayName)


Project_Activity6()

File: Selenium/Project/ProjectActivity6.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.Test;

import java.time.Duration;

public class ProjectActivity6 extends BaseTest{
    @Test
    public void loginTest(){
        wait = new WebDriverWait(driver, Duration.ofSeconds(50));
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[@href='https://alchemy.hguy.co/lms/my-account/']")));

        WebElement myAccountLink = driver.findElement(By.xpath("//a[@href='https://alchemy.hguy.co/lms/my-account/']"));
        myAccountLink.click();
        WebElement loginBtn = driver.findElement(By.xpath("//div/a[@class='ld-login ld-login ld-login-text ld-login-button ld-button']"));
        loginBtn.click();

        WebElement usernametxtBox = driver.findElement(By.xpath("//input[@id='user_login']"));
        WebElement passwordtxtBox = driver.findElement(By.xpath("//input[@id='user_pass']"));
        WebElement signInBtn = driver.findElement(By.xpath(" //input[@id='wp-submit']"));
        usernametxtBox.sendKeys("root");
        passwordtxtBox.sendKeys("pa$$w0rd");
        signInBtn.click();
        WebElement username = driver.findElement(By.xpath("(//a/span[@class='display-name'])[1]"));
        Assert.assertEquals("root",username.getText());

    }
}
